connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -569,8 +559,7 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
                     // Validate the configs we're supposed to update to ensure we're getting a complete configuration
                     // update of all tasks that are expected based on the number of tasks in the commit message.
-                    Map<String, Set<Integer>> updatedConfigIdsByConnector = taskIdsByConnector(deferred);
-                    Set<Integer> taskIdSet = updatedConfigIdsByConnector.get(connectorName);
+                    Set<Integer> taskIdSet = taskIds(deferred);
```

Comment:
If it can't return null, what's the harm in checking?

completeTaskIdSet will still NPE if taskIdSet is null, so I think that being sure is very important.